docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md

```diff
@@ -0,0 +1,211 @@
+           maxWidth: 590,
+         },
+       },
+       {
+         resolve: `gatsby-remark-responsive-iframe`,
+         options: {
+           wrapperStyle: `margin-bottom: 1.0725rem`,
+         },
+       },
+       `gatsby-remark-prismjs`,
+       `gatsby-remark-copy-linked-files`,
+       `gatsby-remark-smartypants`,
+     ],
+   },
+  },
+```
+
+Then, add some options to `gatsby-plugin-feed`. This will allow the the RSS feed of the site to parse MDX.
+
+```diff:title=gatsby-config.js
+- `gatsby-plugin-feed`,
+
++ {
++   resolve: `gatsby-plugin-feed`,
++   options: {
++     query: `
++       {
++         site {
++           siteMetadata {
++             title
++             description
++             siteUrl
++           }
++         }
++       }
++     `,
++     feeds: [
++       {
++         serialize: ({ query: { site, allMdx } }) =>
++           allMdx.edges.map(edge =>
++             Object.assign({}, edge.node.frontmatter, {
++               description: edge.node.excerpt,
++               data: edge.node.frontmatter.date,
++               url: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               guid: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               custom_elements: [{ "content:encoded": edge.node.body }],
++             })
++           ),
++         query: `
++         {
++           allMdx(
++             limit: 1000,
++             sort: { order: DESC, fields: [frontmatter___date] },
++           ) {
++             edges {
++               node {
++                 fields { slug }
++                 frontmatter {
++                   title
++                   date
++                 }
++                 body
++               }
++             }
++           }
++         }
++         `,
++         output: `/rss.xml`,
++         title: `Gatsby RSS feed`,
++       },
++     ],
++   },
++ },
+```
+
+### Step 3
+
+In `gatsby-node.js`, change `allMarkdownRemark` to `allMdx` and `MarkdownRemark` to `Mdx`.
+
+```diff:title=gatsby-node.js
+# line 11
+- allMarkdownRemark(
++ allMdx(
+
+# line 35
+- const posts = result.data.allMarkdownRemark.edges
++ const posts = result.data.allMdx.edges
+
+# line 56
+- if (node.internal.type === `MarkdownRemark`) {
++ if (node.internal.type === `Mdx`) {
+```
+
+### Step 4
+
+In `src/pages/index.js`, replace `allMarkdownRemark` with `allMdx` in both the `render()` method and also the GraphQL query.
+
+```diff:title=src/pages/index.js
+# line 13
+- const posts = data.allMarkdownRemark.edges
++ const posts = data.allMdx.edges
+
+# line 59
+- allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
++ allMdx(sort: { fields: [frontmatter___date], order: DESC }) {
+```
+
+### Step 5
+
+In `src/templates/blog-post.js`, add an import statement for `MDXRenderer` from `gatsby-plugin-mdx` and replace `markdownRemark` with `mdx` in both the `render()` method and also the GraphQL query. Also, we'll be using `body` instead of `html` in our GraphQL query.
```

Comment:
It's the same file, but might make sense to pull out the `body` to `html` change into two substeps. The instinct it to jump to the diff and it's a few different things happening at once.

Reply:
done thanks for this suggestion @laurieontech!